src/web/app/app.tsx

```diff
@@ -0,0 +1,265 @@
+      {props.exp.name()} ({props.exp.weight()}%)
+      <FeatureList
+        title="Enabled features"
+        className="enabled-feature"
+        features={props.exp.enabledFeatures()}
+      />
+      <FeatureList
+        title="Disabled features"
+        className="disabled-feature"
+        features={props.exp.disabledFeatures()}
+      />
+      {paramsList.length > 0 && (
+        <ul className="study-meta">
+          <span>Parameters: </span>
+          {paramsList}
+        </ul>
+      )}
+    </li>
+  );
+}
+
+export function PropertyList(props: {
+  caption: string;
+  list: string[] | null | undefined;
+}) {
+  if (props.list == null || props.list.length === 0) return <></>;
+  return (
+    <ul className="study-meta">
+      <span>{props.caption}: </span>
+      {props.list.map((c) => (
+        <li key={c}>{c}</li>
+      ))}
+    </ul>
+  );
+}
+
+export function IncludeExcludeList(props: {
+  caption: string;
+  include: string[] | null | undefined;
+  exclude: string[] | null | undefined;
+}) {
+  return (
+    <>
+      <PropertyList caption={props.caption} list={props.include} />
+      <PropertyList caption={'Exclude' + props.caption} list={props.exclude} />
+    </>
+  );
+}
+
+export function StudyItem(props: { study: StudyModel; filter: StudyFilter }) {
+  const filter = props.study.filter();
+  const minVersion = props.study.filter()?.min_version ?? '';
+  const maxVersion = props.study.filter()?.max_version ?? '';
+  const experiments = React.useMemo(
+    () => props.study.filterExperiments(props.filter),
+    [props.study, props.filter],
+  );
+  return (
+    <div className="card mb-3">
+      <div className="card-header">
+        <a target="_blank" href={props.study.getConfigUrl()} rel="noreferrer">
+          {props.study.name()}
+        </a>
+      </div>
+      <div className="card-body">
+        <ul className="list-group list-group-flush">
+          {experiments.map((e) => (
+            <ExperimentItem key={e.name()} exp={e} />
+          ))}
+        </ul>
+      </div>
+      <div className="card-footer">
+        {(minVersion !== '' || maxVersion !== '') && (
+          <div className="study-meta">
+            Version range:[ {minVersion !== '' ? minVersion : 'any'} -{' '}
+            {maxVersion !== '' ? maxVersion : 'any'}]
+          </div>
+        )}
+        <PropertyList caption="Channels" list={props.study.channels()} />
+        <PropertyList caption="Platforms" list={props.study.platforms()} />
+        <IncludeExcludeList
+          caption="Countries"
+          include={filter?.country}
+          exclude={filter?.exclude_country}
+        />
+        <IncludeExcludeList
+          caption="Hardware class"
+          include={filter?.hardware_class}
+          exclude={filter?.exclude_hardware_class}
+        />
+      </div>
+    </div>
+  );
+}
+
+export function NavItem(props: { type: SeedType }) {
+  const paramManager = new SearchParamManager(useSearchParams());
```

Comment:
This will get constructed every time the function is run

```suggestion
  const searchParams = useSearchParam()
  const paramManager = React.useMemo(() => new SearchParamManager(searchParams), [searchParams]);
```

But, also, this doesn't seem like it needs to be a separate instance for both `NavItem`'s. In which case, can `App` create one and give it to each `NavItem` in the props?

Reply:
Thanks for the suggestion.
Added search seachParamManager as prop.